source/code/plugins/security_baseline_lib.rb

```diff
@@ -0,0 +1,116 @@
+require 'json'
+require 'securerandom' # SecureRandom.uuid 
+
+require_relative 'oms_common'
+
+module OMS
+    class SecurityBaseline
+
+        @@log = nil
+        def self.log= (value)
+	        @@log = value
+        end
+
+        # ------------------------------------------------------
+        def self.transform_and_wrap(results, host, time)
+            if results == nil
+                @@log.error "Security Baseline Assessment failed; Empty input"
+                return nil, nil
+            end
+
+            if results["results"] == nil 
+                @@log.error "Security Baseline Assessment failed; Invalid input:" + results.inspect
+                return nil, nil
+            end
+
+            results["assessment_id"] = SecureRandom.uuid      
+    
+            asm_baseline_results = results["results"]
+            scan_time = results["scan_time"]
+            assessment_id = results["assessment_id"]
+
+            security_baseline_blob = {
+                "DataType"=>"SECURITY_BASELINE_BLOB", 
+                "IPName"=>"Security",
+                "DataItems"=>[
+                ]
+            }
+
+            asm_baseline_results.each do |asm_baseline_result|
+                oms_baseline_result = self.transform_asm_2_oms(asm_baseline_result, scan_time, host, assessment_id)	
+                security_baseline_blob["DataItems"].push(oms_baseline_result)
+            end 
+
+            security_baseline_summary_blob = self.calculate_summary(results, host, time)
+
+            @@log.info "Security Baseline Summary: " + security_baseline_summary_blob.inspect
+
+            return security_baseline_blob, security_baseline_summary_blob
+        end # self.transform_and_wrap
+    
+        # ------------------------------------------------------   
+        def self.calculate_summary(results, hostname, time)  
+            asm_baseline_results = results["results"]
+            assessment_id = results["assessment_id"]
+            critical_failed_rules = 0;
+            warning_failed_rules = 0;
+            informational_failed_rules = 0;
+            all_failed_rules = 0;
+    
+            asm_baseline_results.each do |asm_baseline_result|
+
+                if asm_baseline_result["result"] == "PASS"  
+                    next 
+                end
+
+                all_failed_rules += 1
+
+                case asm_baseline_result["severity"]
+                when "Critical", "Important"
```

Comment:
OMS Baseline Summary today have only three options: Critical, Warning and Info. 
Adding additional severity level required significant changes in all OMS pipe.